### PR TITLE
Fix conversion of static responses in upstreams

### LINKED ISSUE #745: 404 page not found, after upgrade to latest oauth2-proxy v6.1.0
Hi @JoelSpeed 

I have been using latest image in oauth2-proxy deployment file. 
After the new release, Im seeing '404 page not found', after login and redirect to backend.
I have used the v6.0.0 in the image which solved the issue.

expected behaviour: redirect to backend application after login.
current behaviour: redirect to backend is not happening instead it shows below

image

when accessing app url, It shows login page, after entering the credentials and submit, it shows 404 page not found.

below is oauthproxy deployment
apiVersion: apps/v1
kind: Deployment
metadata:
  labels:
    k8s-app: oauth2-proxy
  name: oauth2-proxy-si
  namespace: myns
spec:
  replicas: 1
  selector:
    matchLabels:
      k8s-app: oauth2-proxy
  template:
    metadata:
      labels:
        k8s-app: oauth2-proxy
    spec:
      containers:
      - args:
        - --provider=oidc
        - --email-domain=*
        - --cookie-domain=.appdomain.com
        - --http-address=0.0.0.0:4180
        - --pass-access-token=true
        - --pass-authorization-header=false
        - --set-authorization-header=true
        - --pass-user-headers=true
        - --set-xauthrequest=true
        - --skip-oidc-discovery=true
        - --login-url=https://idphost:443/oauth2/rest/authorize
        - --insecure-oidc-skip-issuer-verification=true
        - --oidc-issuer-url=https://idphost:443/oauth2
        - --oidc-jwks-url=http://idphost/oauth2/rest/security
        - --redeem-url=http://idphost/oauth2/rest/token
        - --validate-url=http://idphost/oauth2/rest/userinfo
        - --cookie-refresh=1h0m0s
        - --whitelist-domain="idphost"
        - --cookie-httponly=false
        - --cookie-secure=false
        - --standard-logging=true
        - --auth-logging=true
        - --request-logging=true
        - --skip-provider-button=true
        - --upstream=static://200
        - --redis-connection-url=redis://redis-master.default.svc.cluster.local:6379
        - --session-store-type=redis
        env:
        - name: OAUTH2_PROXY_CLIENT_ID
          value: <client id>
        - name: OAUTH2_PROXY_CLIENT_SECRET
          value: <secret>
        - name: OAUTH2_PROXY_COOKIE_SECRET
          value: 
        - name: OAUTH2_PROXY_COOKIE_DOMAIN
          value: .example.com
        image: quay.io/oauth2-proxy/oauth2-proxy:latest
        imagePullPolicy: Always
        name: oauth2-proxy
        ports:
        - containerPort: 4180
          protocol: TCP
---
apiVersion: v1
kind: Service
metadata:
  labels:
    k8s-app: oauth2-proxy
  name: oauthproxy-service-si
  namespace: myns
spec:
  ports:
  - name: http
    port: 4180
    protocol: TCP
    targetPort: 4180
  selector:
    k8s-app: oauth2-proxy
---

using istio envoy filter, let me know if this file is needed

### PR DESCRIPTION
**Pull request by JoelSpeed**
## Description

Fix a bug in the conversion of the static upstream type

## Motivation and Context

Fixes #745 

## How Has This Been Tested?

Manual testing using the local test environment shows this works.

## Checklist:

- [x] My change requires a change to the documentation or CHANGELOG.
- [x] I have updated the documentation/CHANGELOG accordingly.
- [x] I have created a feature (non-master) branch for my PR.

### CODE REVIEW

File: pkg/apis/options/legacy_options_test.go [hunk 1]
```diff
@@ -21,9 +21,10 @@ var _ = Describe("Legacy Options", func() {
 			legacyOpts.LegacyUpstreams.PassHostHeader = true
 			legacyOpts.LegacyUpstreams.ProxyWebSockets = true
 			legacyOpts.LegacyUpstreams.SSLUpstreamInsecureSkipVerify = true
-			legacyOpts.LegacyUpstreams.Upstreams = []string{"http://foo.bar/baz", "file://var/lib/website#/bar"}
+			legacyOpts.LegacyUpstreams.Upstreams = []string{"http://foo.bar/baz", "file://var/lib/website#/bar", "static://204"}
```
Comment (haslersn) [inline pkg/apis/options/legacy_options_test.go:24]: Are these intentionalle only two slashes after `file:`?

Comment (JoelSpeed) [reply pkg/apis/options/legacy_options_test.go:24]: I don't think it particularly matters in this context, but this lead me to find a second related bug, so great spot! I've pushed another commit which I think fixes the conversion of the file based upstreams

File: pkg/apis/options/legacy_options.go [hunk 2]
```diff
@@ -97,17 +100,18 @@ func (l *LegacyUpstreams) convert() (Upstreams, error) {
 			upstream.Static = true
 			upstream.StaticCode = &responseCode
 
-			// These are not allowed to be empty and must be unique
+			// This is not allowed to be empty and must be unique
 			upstream.ID = upstreamString
-			upstream.Path = upstreamString
```
Comment (NickMeves) [inline pkg/apis/options/legacy_options.go:102]: Was this the root of the 404s reported?

**Review by NickMeves — DISMISSED** [review]
I think I follow (I haven't used Envoy + Static myself to have deep insight). LGTM.

File: pkg/apis/options/legacy_options.go (hunk 2 above)
Comment (JoelSpeed) [reply pkg/apis/options/legacy_options.go:102]: Indeed, it was mapping `static://xxx` to the path 🤦‍♂️

**Pull request merged**